core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -310,33 +310,28 @@ object SparkSubmit extends CommandLineUtils {
       RPackageUtils.checkAndBuildRPackage(args.jars, printStream, args.verbose)
     }
 
-    // In client mode, download remote files.
-    if (deployMode == CLIENT) {
-      val hadoopConf = new HadoopConfiguration()
-      args.primaryResource = Option(args.primaryResource).map(downloadFile(_, hadoopConf)).orNull
-      args.jars = Option(args.jars).map(downloadFileList(_, hadoopConf)).orNull
-      args.pyFiles = Option(args.pyFiles).map(downloadFileList(_, hadoopConf)).orNull
-      args.files = Option(args.files).map(downloadFileList(_, hadoopConf)).orNull
-    }
+    val hadoopConf = new HadoopConfiguration()
+    val targetDir = Files.createTempDirectory("tmp").toFile
+    val sparkConf = new SparkConf(loadDefaults = false).setAll(args.sparkProperties)
```

Comment:
@vanzin yes, your concern is correct. `Logging` module will be initialized by other class, thus default log level constraint is broken.

The purpose here is to use `Utils.doFetchFile` without duplication. One solution I could think of is to add the similar code here to avoid logging initialization.